Utilities/Files.cs

```diff
@@ -237,4 +277,53 @@ public static bool IsFileLocked(FileInfo file)
             return false;
         }
     }
+
+    // Code from https://www.codeproject.com/Tips/1190802/File-Locking-in-a-Multi-Threaded-Environment
+    // The code maintains a concurrent dictionary of lock objects and keeps track of how many calls are in the queue for
+    // accessing each filename. If the count goes down to 0, the key is removed from the dictionary as a cleanup mechanism.
+    class LockManager
+    {
+        private static readonly ConcurrentDictionary<string, lobj> _locks =
+                new ConcurrentDictionary<string, lobj>();
+
+        public static void GetLock(string filename, Action action)
```

Comment:
NIT - this class is generic. I'd use `lockname` instead of `filename`

Reply:
It's not a public class. It's internal to Files.cs. While it could be extracted as a generic class, that is not the way we are currently using it.

Reply:
I'll go ahead and move it to a public class so that it can be used a general utility.